src/forms/inputs/file-input/file-input.js

```diff
@@ -48,107 +60,195 @@ import classnames from 'classnames'
  */
 
 const propTypes = {
-  ...fieldPropTypes,
-  onLoad: PropTypes.func,
+  ...fileInputPropTypes,
   thumbnail: PropTypes.string,
   hidePreview: PropTypes.bool,
   className: PropTypes.string,
   previewComponent: PropTypes.func,
   children: PropTypes.node,
+  multiple: PropTypes.bool,
+  onRemove: PropTypes.func,
+  readFiles: PropTypes.func,
+  removeComponent: PropTypes.func,
+  selectText: PropTypes.string,
 }
 
 const defaultProps = {
-  onLoad: noop,
+  hidePreview: false,
+  multiple: false,
+  onRemove: noop,
+  readFiles: readFilesAsDataUrls,
+  removeComponent: RemoveButton,
+  selectText: '',
 }
 
-// Promise wrapper around FileReader
-function readFileAsDataUrl (file) {
-  // eslint-disable-next-line no-undef
-  const fileReader = new window.FileReader()
-  return new Promise((resolve, reject) => {
-    // Resolve with file data on success
-    fileReader.onload = (event) => resolve(event.target.result)
-    fileReader.onerror = (error) => reject(error)
-    return fileReader.readAsDataURL(file)
+function usePrevious(value) {
+  const ref = useRef()
+  useEffect(() => {
+    ref.current = value
   })
+  return ref.current
 }
 
-class FileInput extends React.Component {
+function FileInput(props) {
+  const {
+    input,
+    meta,
+    className, // eslint-disable-line no-unused-vars
+    submitting,
+    accept,
+    capture,
+    hidePreview,
+    multiple,
+    readFiles,
+    removeComponent: RemoveComponent,
+    selectText,
+    thumbnail,
+    onRemove,
+    ...rest
+  } = omitLabelProps(props)
 
-  constructor (props) {
-    super(props)
-    this.state = { file: null }
-    this.loadFile = this.loadFile.bind(this)
-    this.onChange = this.onChange.bind(this)
-  }
+  const [errors, setErrors] = useState(null)
+  const inputRef = useRef()
+  const prevMultiple = usePrevious(multiple)
 
-  loadFile (e) {
-    // Don't attempt file reading in SSR mode
-    if (isServer()) return
-    // Read file as data URL and call change handlers
-    const file = e.target.files[0]
-    return readFileAsDataUrl(file)
-      .then((fileData) => this.onChange(fileData, file))
+  const clearFileInput = () => {
+    /* istanbul ignore next */
+    if (inputRef.current) {
+      inputRef.current.value = ''
+    }
   }
 
-  onChange (fileData, file) {
-    // Call redux forms onChange and onLoad callback
-    const { input: { onChange }, onLoad } = this.props
-    onChange(fileData)
-    onLoad(fileData, file)
-    this.setState({ file })
-  }
+  const removeFile = useCallback(async (idx) => {
+    const { onChange, value } = input
+    const [removedFile, remainingFiles] = removeAt(value, idx)
+
+    try {
+      await onRemove(removedFile)
+
+      // If all files have been removed, then reset the native input
+      if (!remainingFiles.length) clearFileInput()
+
+      // This method is only available when multiple="true", so always set an array
```

Comment:
Is this comment still applicable (since we're now **always** setting the value to an array)?